tasmota/xsns_02_analog.ino

```diff
@@ -334,6 +370,32 @@ uint16_t AdcGetLux(uint32_t idx) {
   return (uint16_t)ldrLux;
 }
 
+float AdcGetMq(uint32_t idx) {
+  float avg = 0.0;
+  int retries = 5;
+  int retries_delay = 20;
+  float _RL = 10; //Value in KiloOhms
+  float _R0 = 10;
+  int _adc = AdcRead(Adc[idx].pin, 2);
+  avg += _adc;
+  for (int i = 0; i < retries-1; i ++) {
+    delay(retries_delay);
+    _adc = AdcRead(Adc[idx].pin, 2);
+    avg += _adc;
+  }
+  float voltage = (avg/ retries) * ANALOG_V33 / ((pow(2, ANALOG_RESOLUTION)) - 1);
```

Comment:
You are assuming a 3V3 range which is true on a ESP8266 Wemos D1 / NodeMCU due to the existing voltage divider.
Will be false on an different board with different consideration such as a bare ESP module.

I'm not sure that the range is the same on ESP32 DevKit or various ESP32 based boards.

Reply:
Suggest?

Reply:
ANALOG_V33 const is also used for other analog reader

Reply:
There is 1 use case in `AdcEverySecond()` where there is a #ifdef ESP8266 ...

You have checked both box that the PR has been tested on ESP8266 and on ESP-32
Is that the case ?
On which boards have you tested ?
Can you garantue that this PR will work exactly the same on standard ESP8266 and ESP-32 boards ?

Reply:
i tried with wemos d1 mini and esp32 dev kit

Reply:
Wemos d1 mini supports 3.2v max voltage
https://www.wemos.cc/en/latest/d1/d1_mini.html

Reply:
I don't know esp board that works with max voltage other than 3,3v

Reply:
Esp8266.... 0..1V

Reply:
Ok how could i fix them?

Reply:
THe problem is you can't use mq if you have max voltage of 1V!

Reply:
@barbudor how could we do with 1V ADC?

Reply:
It's ok, most people will use a Wemos or NodeMCU
Just add in the beginning of the code some pseudo wiring diagram like for the other sensors
And when you will PR the docs, don't forget to add that it needs a 3V3 input like NodeMCU/WemosD1
Thanks

Reply:
Done, thanks